crm_phonecall/report/crm_phonecall_report.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# Copyright 2004-2010 Tiny SPRL (<http://tiny.be>)
+# Copyright 2017 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import tools
+from openerp import fields, models
+
+AVAILABLE_STATES = [
+    ('draft', 'Draft'),
+    ('open', 'Todo'),
+    ('cancel', 'Cancelled'),
+    ('done', 'Held'),
+    ('pending', 'Pending')
+]
+
+
+class CrmPhonecallReport(models.Model):
+    _name = "crm.phonecall.report"
+    _description = "Phone calls by user"
+    _auto = False
+
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='User',
+        readonly=True,
+    )
+    team_id = fields.Many2one(
+        comodel_name='crm.team',
+        string='Team',
+        readonly=True,
+    )
+    priority = fields.Selection(
+        selection=[
+            ('0', 'Low'),
+            ('1', 'Normal'),
+            ('2', 'High')
+        ],
+        string='Priority',
+    )
+    nbr_cases = fields.Integer(
+        string='# of Cases',
+        readonly=True,
+    )
+    state = fields.Selection(
+        AVAILABLE_STATES,
+        string='Status',
+        readonly=True,
+    )
+    create_date = fields.Datetime(
+        string='Create Date',
+        readonly=True,
+        select=True,
+    )
+    delay_close = fields.Float(
+        string='Delay to close',
+        digits=(16, 2),
+        readonly=True,
+        group_operator="avg",
+        help="Number of Days to close the case",
+    )
+    duration = fields.Float(
+        string='Duration',
+        digits=(16, 2),
+        readonly=True,
+        group_operator="avg",
+    )
+    delay_open = fields.Float(
+        string='Delay to open',
+        digits=(16, 2),
+        readonly=True,
+        group_operator="avg",
+        help="Number of Days to open the case",
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        readonly=True,
+    )
+    opening_date = fields.Datetime(
+        readonly=True,
+        select=True,
+    )
+    date_closed = fields.Datetime(
+        string='Close Date',
+        readonly=True,
+        select=True)
+
+    def init(self, cr):
+
+        tools.drop_view_if_exists(cr, 'crm_phonecall_report')
+        cr.execute("""
+            create or replace view crm_phonecall_report as (
```

Comment:
Make this report inheritable extracting the SQL parts in methods as you already seen in for example invoices.